File: test/csharp/SpecGateFixtures/MultiFile/Greet.cs
using SpecGate.Annotations;

namespace SpecGateFixtures.MultiFile;

/// <summary>
/// Greeting fixture used by the multi-file scenario, paired with
/// <see cref="FarewellOps"/> to prove operations split across files are all
/// discovered and compiled together.
/// </summary>
public static class GreetOps
{
    /// <summary>Builds a greeting for <paramref name="name"/>.</summary>
    /// <param name="name">The name to greet (spec input <c>name</c>).</param>
    /// <returns>The string <c>"Hello, {name}!"</c>.</returns>
    [SpecOperation("greet")]
    public static string Greet([SpecInput("name")] string name) => $"Hello, {name}!";
}

File: test/rust/crates/specgate-fixtures/src/conformance/multi_file/greet.rs
// Operations split across multiple files — tests that the harness
// scans all .rs files, not just one.

use specgate::*;

#[spec_operation("greet", spec = "fixture.multi_file")]
pub fn greet(name: &str) -> String {
    format!("Hello, {name}!")
}
